proxy/proxy.go

```diff
@@ -42,9 +48,23 @@ type GetProxyURLs func() []string
 // which will proxy requests to an etcd cluster.
 // The handler will periodically update its view of the cluster.
 func NewHandler(t *http.Transport, urlsFunc GetProxyURLs, failureWait time.Duration, refreshInterval time.Duration) http.Handler {
+	proxyURLs := make(map[string]struct{})
```

Comment:
I use the type `map[string]struct{}` to make it same as the one in `director` so that the director can use its `banned` which is also `map[string]struct{}` to remove cyclic proxy URLs from its endpoints (when calling `refresh` or `endpoints` method).